modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/conn/ClientConnFactory.java

```diff
@@ -81,8 +81,17 @@ public ClientConnFactory(
     
     private SSLContext getSSLContext(final IOSession iosession) {
         InetSocketAddress address = (InetSocketAddress) iosession.getRemoteAddress();
-        String host = address.getHostName() + ":" + address.getPort();
-        return  getSSLContextForHost(host);
```

Comment:
Shall we pass the address variable to getSSLContextForHost method and modify it there . It seems that is being used in some other place too.